### PR TITLE
Improves UI redirection after successful authentication

### PR DESCRIPTION
**Pull request by pratap0007**
This patch improves ui redirection by redirecting users to the same page
where they had attempted to login after successful authentication

Signed-off-by: Shiv Verma <shverma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Run API Unit Tests, Lint Checks, API Design, Golden Files with `make api-check`
- [ ] Run UI Unit Tests, Lint Checks with `make ui-check`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: ui/src/containers/Header/index.tsx [hunk 1]
```diff
@@ -31,13 +31,55 @@ import { AUTH_BASE_URL, REDIRECT_URI } from '../../config/constants';
...
 
 const Header: React.FC = observer(() => {
   const { user, providers } = useMst();
   const history = useHistory();
   const [isModalOpen, setIsModalOpen] = React.useState(false);
   const [disable, setDisable] = React.useState(false);
 
+  const detailPagePath = `detailPagePath`;
+
+  useEffect(() => {
```
Comment (PuneetPunamiya) [inline ui/src/containers/Header/index.tsx:44]: Any reason we are moving and handling all the code to header and not parser url

Comment (pratap0007) [reply ui/src/containers/Header/index.tsx:44]: these code are related to login so I had moved it to header as parse url component manage parsing of url params

Comment (PuneetPunamiya) [reply ui/src/containers/Header/index.tsx:44]: Since it is related to login, then shouldn't we move this to login component ?

Comment (pratap0007) [reply ui/src/containers/Header/index.tsx:44]: we don't have login component and handling login within header component

**Pull request closed without merging**